specify/specify7
# Ensure WorkBench always checks for uniqueness rules

Fixes #4500 


### Checklist

      and self-explanatory (or properly documented)

### Testing instructions
For any table:
- Ensure the table has at least one other uniqueness rule
- Create a new uniqueness rule which has at least one different Unique Field
- Create or use a WorkBench Dataset which contains the fields for the created uniqueness rule, but does not contain the field for other uniqueness rule(s) for the Table 
  - Alternatively, fields from other uniqueness rule(s) can be used, but ensure the value for these fields in the Dataset is empty (blank)
- Try and invalidate the created uniqueness rule by having more than one row which would violate the uniqueness rule

For an example of this, see https://github.com/specify/specify7/issues/4500#issue-2127446816. 
For another example, I used the following Uniqueness Rules on the Agent table for testing and created a Dataset with columns which mapped to `Last Name`, `Email`, and `Abbreviation`):

<img width="543" alt="Screenshot 2024-02-28 at 9 38 03 AM" src="https://github.com/specify/specify7/assets/64045831/b0e4bac3-1bc9-40cc-8012-045c5ef54199">
